consensus/spos/bls/subroundEndRound.go

```diff
@@ -553,6 +565,119 @@ func (sr *subroundEndRound) aggregateSigsAndHandleInvalidSigners(bitmap []byte)
 	return bitmap, sig, nil
 }
 
+func (sr *subroundEndRound) checkGoRoutinesThrottler(ctx context.Context) error {
+	for {
+		if sr.signatureThrottler.CanProcess() {
+			break
+		}
+		select {
+		case <-time.After(time.Millisecond):
+			continue
+		case <-ctx.Done():
+			return spos.ErrTimeIsOut
+		}
+	}
+	return nil
+}
+
+func (sr *subroundEndRound) signatureVerification(wg *sync.WaitGroup, i int, pk string, invalidPubKey *[]string, mutex *sync.Mutex, sigShare []byte, mutexBool *sync.Mutex, errorReturned *error) {
```

Comment:
please refactor, removing the wg, invalidPubKey, mutex,  mutexBool, errorReturned from the arguments list.
you can do the rest in an anonymous function that is called on a goroutine.